### PR TITLE
Add Pos#withLookAt

### PR DESCRIPTION
**Pull request by ZakShearman**
Add a withLookAt method to Pos based on code from Entity#lookAt

### CODE REVIEW

**Comment by ZakShearman** [comment]
Not sure if I got the unit test naming right and also, should the Entity lookAt unit tests be changed/removed since they are almost the same as the new Pos ones.

File: src/main/java/net/minestom/server/coordinate/Pos.java [hunk 1]
```diff
@@ -112,6 +113,19 @@ public Pos(@NotNull Point point) {
         return new Pos(x, y, z, yaw, pitch);
     }
 
+    @Contract(pure = true)
+    public @NotNull Pos withLookAt(@NotNull Pos position) {
```
Comment (Marlonzockt) [inline src/main/java/net/minestom/server/coordinate/Pos.java:117]: The parameter could be a Point instead of a Pos, couldn't it?

Comment (ZakShearman) [reply src/main/java/net/minestom/server/coordinate/Pos.java]: Indeed, changed this now.

File: src/main/java/net/minestom/server/coordinate/Pos.java [hunk 2]
```diff
@@ -112,6 +113,19 @@ public Pos(@NotNull Point point) {
...
     }
 
+    @Contract(pure = true)
+    public @NotNull Pos withLookAt(@NotNull Point point) {
+        if (this.samePoint(point)) {
+            return this;
+        }
+
+        Vec delta = Vec.fromPoint(point.sub(this)).normalize();
+        return new Pos(x, y, z,
```
[the remark this answers is not in the document]
Comment (ZakShearman) [reply src/main/java/net/minestom/server/coordinate/Pos.java]: Thanks I've changed this 👍🏻

**Pull request merged**